From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nikolajs Agafonovs <nikolajs.agafonovs@gmail.com>
Date: Wed, 21 Oct 2026 16:20:08 +0300
Subject: [PATCH] x86_64-asm: rdtscp, clac, stac, xgetbv, xsetbv, monitor,
 mwait, rdrand, rdseed, rdpid

New table entries (0F 01 group, 0F C7 /6 and /7 with the 16/32/64-bit
register forms, F3 0F C7 /7).  All encodings byte-compared with GNU as.
---
 x86_64-asm.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/x86_64-asm.h b/x86_64-asm.h
index d671243..93b38c8 100644
--- a/x86_64-asm.h
+++ b/x86_64-asm.h
@@ -44,7 +44,21 @@
     DEF_ASM_OP0L(vmlaunch, 0xc2, 0, OPC_0F01)
     DEF_ASM_OP0L(vmresume, 0xc3, 0, OPC_0F01)
     DEF_ASM_OP0L(vmxoff, 0xc4, 0, OPC_0F01)
+    DEF_ASM_OP0L(monitor, 0xc8, 0, OPC_0F01)
+    DEF_ASM_OP0L(mwait, 0xc9, 0, OPC_0F01)
+    DEF_ASM_OP0L(clac, 0xca, 0, OPC_0F01)
+    DEF_ASM_OP0L(stac, 0xcb, 0, OPC_0F01)
+    DEF_ASM_OP0L(xgetbv, 0xd0, 0, OPC_0F01)
+    DEF_ASM_OP0L(xsetbv, 0xd1, 0, OPC_0F01)
     DEF_ASM_OP0L(swapgs, 0xf8, 0, OPC_0F01)
+    DEF_ASM_OP0L(rdtscp, 0xf9, 0, OPC_0F01)
+    DEF_ASM_OP1(rdrand, 0x0fc7, 6, OPC_MODRM, OPT_REG32)
+ALT(DEF_ASM_OP1(rdrand, 0x660fc7, 6, OPC_MODRM, OPT_REG16))
+ALT(DEF_ASM_OP1(rdrand, 0x0fc7, 6, OPC_MODRM | OPC_48, OPT_REG64))
+    DEF_ASM_OP1(rdseed, 0x0fc7, 7, OPC_MODRM, OPT_REG32)
+ALT(DEF_ASM_OP1(rdseed, 0x660fc7, 7, OPC_MODRM, OPT_REG16))
+ALT(DEF_ASM_OP1(rdseed, 0x0fc7, 7, OPC_MODRM | OPC_48, OPT_REG64))
+    DEF_ASM_OP1(rdpid, 0xf30fc7, 7, OPC_MODRM, OPT_REG64)
 
      /* strings */
 ALT(DEF_ASM_OP0L(cmpsb, 0xa6, 0, OPC_BWLX))
